graphjet-core/src/main/java/com/twitter/graphjet/bipartite/api/EdgeMetadataIntIterator.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2017 Twitter. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.twitter.graphjet.bipartite.api;
+
+import it.unimi.dsi.fastutil.ints.IntIterator;
+
+/**
+ * An interface accessing the edge metadata {@link IntIterator}.
+ */
+public interface EdgeMetadataIntIterator extends IntIterator {
```

Comment:
This interface is an int edge iterator with access to edge metadata, not an edge metadata iterator. 
The original class name meant to say IntIterator with access to edge metadata. Now I change the name to WithEdgeMetadataIntIterator to make it more clear.

The reason that this has to be an IntIterator is, in Segment level, we need an IntIterator with access to edge metadata. See the changes in InternalIdToLongIterator.java in this rb.